shared/api/deployments.go

```diff
@@ -0,0 +1,254 @@
+package api
+
+// DeploymentsPost represents the fields of a new LXD deployment
+//
+// swagger:model
+//
+// API extension: deployments.
+type DeploymentsPost struct {
+	DeploymentPost `yaml:",inline"`
+	DeploymentPut  `yaml:",inline"`
+}
+
+// DeploymentPost represents the fields required to rename a LXD deployment
+//
+// swagger:model
+//
+// API extension: deployments.
+type DeploymentPost struct {
+	// The name for the deployment
+	// Example: myapp
+	Name string `json:"name" yaml:"name"`
+}
+
+// DeploymentPut represents the modifiable fields of a LXD deployment
+//
+// swagger:model
+//
+// API extension: deployments.
+type DeploymentPut struct {
+	// Description of the deployment
+	// Example: My new app and its required services
+	Description string `json:"description" yaml:"description"`
+
+	// Deployment configuration map (refer to doc/deployments.md)
+	// Example: {"user.mykey": "foo"}
+	Config map[string]string `json:"config" yaml:"config"`
+
+	// DeploymentShapes keyed by name
+	// Example: k8s-kubelet (map key)
+	DeploymentShapes map[string]DeploymentShape `json:"deployment_shapes" yaml:"deployment_shapes"`
```

Comment:
This seems like stuttering to include "Deployment" in this field, can we just call it Shapes?